### PR TITLE
dns/eve: add 'HTTPS' type logging

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/4751) ticket.

Previous PR: https://github.com/OISF/suricata/pull/8031

Describe new changes:

- Added logic to parse rrtype: HTTPS
-  Modified output-json-dns.c file

suricata-verify-pr: 969

### CODE REVIEW

**Comment by hsadia538** [comment]
I created a Suricata verify test to verify the above. Do I need to add the test details somewhere as well? I was able to verify the output correctly.

**Comment by inashivb** [comment]
> I created a Suricata verify test to verify the above. Do I need to add the test details somewhere as well? I was able to verify the output correctly.

Woohooo. Congratulations! 🥳 
Yes, please create a PR for the suricata-verify test in the suricata-verify repo and link the s-v PR number in the PR description here. e.g. https://github.com/OISF/suricata/pull/8042 (Notice `suricata-verify-pr` in the PR description)

File: src/output-json-dns.c [hunk 1]
```diff
@@ -187,66 +189,21 @@ typedef enum {
...
+    { "rt", LOG_RT }, { "nsap", LOG_NSAP }, { "nsapptr", LOG_NSAPPTR }, { "sig", LOG_SIG },
+    { "key", LOG_KEY }, { "px", LOG_PX }, { "gpos", LOG_GPOS }, { "aaaa", LOG_AAAA },
+    { "loc", LOG_LOC }, { "nxt", LOG_NXT }, { "srv", LOG_SRV }, { "atma", LOG_ATMA },
+    { "naptr", LOG_NAPTR }, { "kx", LOG_KX }, { "cert", LOG_CERT }, { "a6", LOG_A6 },
+    { "dname", LOG_DNAME }, { "opt", LOG_OPT }, { "apl", LOG_APL }, { "ds", LOG_DS },
+    { "sshfp", LOG_SSHFP }, { "ipseckey", LOG_IPSECKEY }, { "rrsig", LOG_RRSIG },
+    { "nsec", LOG_NSEC }, { "dnskey", LOG_DNSKEY }, { "dhcid", LOG_DHCID }, { "nsec3", LOG_NSEC3 },
+    { "nsec3param", LOG_NSEC3PARAM }, { "tlsa", LOG_TLSA }, { "hip", LOG_HIP }, { "cds", LOG_CDS },
+    { "cdnskey", LOG_CDNSKEY }, { "https", LOG_HTTPS }, { "spf", LOG_SPF }, { "tkey", LOG_TKEY },
+    { "tsig", LOG_TSIG }, { "maila", LOG_MAILA }, { "any", LOG_ANY }, { "uri", LOG_URI } };
```
Comment (inashivb) [inline src/output-json-dns.c:206]: Was this clang-format's doing? 😢

**Review by inashivb — APPROVED** [review]
Great job, @hsadia538 !
This tests ok on my end. LGTM. Thanks! :)

Edit: Please note that we can only accept the PR once you have updated the suricata-verify test to go with it.

File: src/output-json-dns.c (hunk 1 above)
Comment (hsadia538) [reply src/output-json-dns.c:206]: Yes. I believe it looked nicer before. But wasn't sure if I should revert back or not.

**Comment by hsadia538** [comment]
> Great job, @hsadia538 ! This tests ok on my end. LGTM. Thanks! :)
> 
> Edit: Please note that we can only accept the PR once you have updated the suricata-verify test to go with it.

Alright. I will update here when I have the test ready.Thanks

**Comment by hsadia538** [comment]
> Woohooo. Congratulations! 🥳 Yes, please create a PR for the suricata-verify test in the suricata-verify repo and link the s-v PR number in the PR description here. e.g. #8042 (Notice `suricata-verify-pr` in the PR description)

I have updated the Suricata Verify PR number in the description.

**Comment by jufajardini** [comment]
> > Woohooo. Congratulations! partying_face Yes, please create a PR for the suricata-verify test in the suricata-verify repo and link the s-v PR number in the PR description here. e.g. #8042 (Notice `suricata-verify-pr` in the PR description)
> 
> I have updated the Suricata Verify PR number in the description.

Thanks, I'll see if we manage to make the CI checks run with that PR :)
If not, a new PR will be required, to trigger the CI checks with the right tests.

**Comment by hsadia538** [comment]
> > > Woohooo. Congratulations! partying_face Yes, please create a PR for the suricata-verify test in the suricata-verify repo and link the s-v PR number in the PR description here. e.g. #8042 (Notice `suricata-verify-pr` in the PR description)
> > 
> > 
> > I have updated the Suricata Verify PR number in the description.
> 
> Thanks, I'll see if we manage to make the CI checks run with that PR :) If not, a new PR will be required, to trigger the CI checks with the right tests.

Ok sure , let me know.

File: src/output-json-dns.c (hunk 1 above)
Comment (victorjulien) [reply src/output-json-dns.c:206]: yes please revert this, then wrap it in:
```c
// clang-format off
... the table ...
// clang-format on
```

**Review by victorjulien — CHANGES_REQUESTED** [review]
clang-format issue

File: src/output-json-dns.c (hunk 1 above)
Comment (hsadia538) [reply src/output-json-dns.c]: done

**Comment by hsadia538** [comment]
followed by #8076

**Pull request closed without merging**